spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfigurationTests.java

```diff
@@ -525,67 +489,77 @@ void initSqlsWithFlywayUrl() {
-					assertThat(context).hasFailed();
-					Throwable failure = context.getStartupFailure();
-					assertThat(failure).hasRootCauseInstanceOf(FlywayTeamsUpgradeRequiredException.class);
-					assertThat(failure).hasMessageContaining(" oracle.kerberosCacheFile ");
-				});
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException(
+				"spring.flyway.oracle-kerberos-cache-file=/tmp/cache", "oracle.kerberosCacheFile");
 	}
 
 	@Test
 	void oracleKerberosConfigFileIsCorrectlyMapped() {
-		this.contextRunner.withUserConfiguration(EmbeddedDataSourceConfiguration.class)
-				.withPropertyValues("spring.flyway.oracle-kerberos-config-file=/tmp/config").run((context) -> {
-					assertThat(context).hasFailed();
-					Throwable failure = context.getStartupFailure();
-					assertThat(failure).hasRootCauseInstanceOf(FlywayTeamsUpgradeRequiredException.class);
-					assertThat(failure).hasMessageContaining(" oracle.kerberosConfigFile ");
-				});
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException(
+				"spring.flyway.oracle-kerberos-config-file=/tmp/config", "oracle.kerberosConfigFile");
 	}
 
 	@Test
 	void outputQueryResultsIsCorrectlyMapped() {
-		this.contextRunner.withUserConfiguration(EmbeddedDataSourceConfiguration.class)
-				.withPropertyValues("spring.flyway.output-query-results=false").run((context) -> {
-					assertThat(context).hasFailed();
-					Throwable failure = context.getStartupFailure();
-					assertThat(failure).hasRootCauseInstanceOf(FlywayTeamsUpgradeRequiredException.class);
-					assertThat(failure).hasMessageContaining(" outputQueryResults ");
-				});
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.output-query-results=false",
+				"outputQueryResults");
 	}
 
 	@Test
 	void skipExecutingMigrationsIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.skip-executing-migrations=true",
+				"skipExecutingMigrations");
+	}
+
+	@Test
+	void conjurUrlIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException(
+				"spring.flyway.conjur-url=http://foo.com/secrets", "conjurUrl");
+	}
+
+	@Test
+	void conjurTokenIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.conjur-token=5150",
+				"conjurToken");
+	}
+
+	@Test
+	void vaultUrlIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.vault-url=http://foo.com/secrets",
+				"vaultUrl");
+	}
+
+	@Test
+	void vaultTokenIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.vault-token=5150", "vaultToken");
+	}
+
+	@Test
+	void vaultSecretsIsCorrectlyMapped() {
+		assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException("spring.flyway.vault-secrets=kv/test/1/config",
+				"vaultSecrets");
+	}
+
+	@Test
+	private void assertThatPropertyResultsInFlywayTeamsUpgradeRequiredException(String propertyKeyAndValue,
```

Comment:
I'd prefer that the setup remains in each test and that the thing that's shared is the real assertion in the form of a `ContextConsumer`. 

A good example of what I am trying to convey is  in [`EnvironmentEndpointAutoConfigurationTests`](https://github.com/spring-projects/spring-boot/blob/1bd603f3aa48f29807db19dc85d9a7140214df0c/spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/env/EnvironmentEndpointAutoConfigurationTests.java#L62-L87).